### PR TITLE
perf: support new api engine_opSealPayload and optimize overheads of block mining

### PR DESCRIPTION
**Pull request by bnoieh**
### Description

This PR adds two changes to op-geth:
1. support new api engine_opSealPayload, this is the associated PR to https://github.com/bnb-chain/opbnb/pull/248
2. optimize several overheads in block mining to achieve higher TPS
### Rationale
Through load test, we found that it took about 200ms(10k tps case) to call/serve of engine apis and some overheads in block minting steps can be optimized. So we try to optimize it to achieve higher TPS.

### Example

n/a

### Changes

Notable changes:
* support new api engine_opSealPayload
```
# original block produce flow
1. op-node call engine_forkchoiceUpdated
2. wait scheduled time
3. op-node call sequentially: engine_getPayload -> engine_newPayload -> engine_forkchoiceUpdated

# new block produce flow using new engine_opSealPayload API
1. op-node call engine_forkchoiceUpdated
2. wait scheduled time
3. op-node call concurrently: engine_opSealPayloadResponse & engine_getPayload
```
* optimized points in block mining steps
```
1. Build / FinalizeAndAssemble step: concurrent IntermediateRoot   and NewBlockWithWithdrawals
2. NewPayload / ExecutableDataToBlock step: cache Block in GetPayload API then get from cache instead of ExecutableDataToBlock
3. NewPayload / it.next(validateHead inside) step:  skip it.next if in minerMode
4. NewPayload / ValidateState step: async running ValidateState
5. ForkchoiceUpdate Head / getBlockByHash step: cache block in NewPayload then get from cache
6. ForkchoiceUpdate Head / setCanoncial step: concurrent writeHeadBlock and collectLogs
7. NewPayload / report statistics: skip report snapshot/pathdb size if in minerMode
```

### CODE REVIEW

File: core/blockchain.go [hunk 1]
```diff
@@ -1731,6 +1738,15 @@ func (bc *BlockChain) insertChain(chain types.Blocks, setHead bool) (int, error)
 		return 0, nil
 	}
 
+	minerMode := false
```
Comment (andyzhang2023) [inline core/blockchain.go:1741]: Should we use --mine flag to decide the value of minerMode?

**Review by andyzhang2023 — APPROVED** [review]
LGTM

**Review by krish-nr — APPROVED** [review]
LGTM

**Review by welkin22 — APPROVED** [review]
LGTM

**Pull request merged**